pkg/validation/application_definition.go

```diff
@@ -58,6 +58,27 @@ func ValidateApplicationDefinitionUpdate(newAd appskubermaticv1.ApplicationDefin
 	return allErrs
 }
 
+func ValidateApplicationDefinitionDelete(ad appskubermaticv1.ApplicationDefinition) field.ErrorList {
+	labels := ad.GetLabels()
+	if labels != nil {
+		if _, exists := labels[appskubermaticv1.ApplicationManagedByLabel]; exists {
```

Comment:
I think you should also check the value of the label and make sure it's `appskubermaticv1.ApplicationManagedByKKPValue` i.e. "kkp"